Dockerfile

```diff
@@ -12,9 +12,13 @@ COPY . .
 RUN pip3 install poetry
 Run poetry install
 
+
+FROM base as development
 CMD [ "poetry", "run" , "flask", "run", "--host=0.0.0.0"]
 
-#EXPOSE 8084
-#RUN chmod +x ./run_gunicorn.sh
-#ENTRYPOINT ["./run_gunicorn.sh"]
+FROM base as production 
+RUN pip3 install gunicorn flask
+#CMD [ "poetry", "run", "gunicorn", "--bind 0.0.0.0 "todo_app.app:create_app()"]
+RUN chmod +x ./run_gunicorn.sh
+ENTRYPOINT ["./run_gunicorn.sh"]
```

Comment:
This seems to be _really_ close, only two things I needed to adjust:
* Making sure the bash script had the right line endings (LF) - this might not be an issue for you (depends on how you're accessing the code) but could be one to watch out for
* The actual script needs updating from `poetry run gunicorn --bind 0.0.0.0 "todo_app.app:create_app()"` to `poetry run gunicorn --bind 0.0.0.0 "todo_app.app:app"` since you haven't yet completed M3 so we haven't switched to use the create_app function yet

With those changes, and the update to make sure the run command actually sets the right options, your prod container was working for me! 🙂